rpc-server/src/dispatchers/blockchain.rs

```diff
@@ -529,4 +588,152 @@ impl BlockchainInterface for BlockchainDispatcher {
+                        block_hash,
+                        block_number,
+                        mut tx_logs,
+                    } => {
+                        let mut block_log = None;
+                        inherent_logs.retain(|log| is_log_related_to_addresses(&log, &addresses));
+                        tx_logs.retain(|tx_log| {
+                            is_tx_logs_related_to_any_addresses(&tx_log, &addresses)
+                        });
+                        if !inherent_logs.is_empty() || !tx_logs.is_empty() {
+                            block_log = Some(BlockLog::AppliedBlockLog {
+                                inherent_logs,
+                                block_hash,
+                                block_number,
+                                tx_logs,
+                            });
+                        }
+                        block_log
+                    }
+                };
+                future::ready(result)
+            })
+            .boxed())
+    }
+
+    /// Subscribes to log events related to a given list of addresses.
+    #[stream]
+    async fn logs_by_type_subscribe(
+        &mut self,
+        log_types: Vec<LogType>,
+    ) -> Result<BoxStream<'static, BlockLog>, Self::Error> {
+        let stream = self.logs_subscribe().await?;
+
+        Ok(stream
+            .filter_map(move |event| {
+                let result = match event {
+                    BlockLog::AppliedBlockLog {
+                        mut inherent_logs,
+                        block_hash,
+                        block_number,
+                        mut tx_logs,
+                    } => {
+                        let mut block_log = None;
+                        inherent_logs.retain(|log| contains_log_type(&log, &log_types));
+                        tx_logs.retain(|tx_log| tx_logs_contains_any_log_type(&tx_log, &log_types));
+                        if !inherent_logs.is_empty() || !tx_logs.is_empty() {
+                            block_log = Some(BlockLog::AppliedBlockLog {
+                                inherent_logs,
+                                block_hash,
+                                block_number,
+                                tx_logs,
+                            });
+                        }
+                        block_log
+                    }
+                    BlockLog::RevertBlockLog {
+                        mut inherent_logs,
+                        block_hash,
+                        block_number,
+                        mut tx_logs,
+                    } => {
+                        let mut block_log = None;
+                        inherent_logs.retain(|log| contains_log_type(&log, &log_types));
+                        tx_logs.retain(|tx_log| tx_logs_contains_any_log_type(&tx_log, &log_types));
+                        if !inherent_logs.is_empty() || !tx_logs.is_empty() {
+                            block_log = Some(BlockLog::AppliedBlockLog {
```

Comment:
Same here, this is in the `RevertBlockLog` match block:
```suggestion
                            block_log = Some(BlockLog::RevertBlockLog {
```